### PR TITLE
Initial Apply job changes

### PR DESCRIPTION
**Pull request by Himan10**
Fixes: 
1. Might be having some template rendering issues while calling jobs/company APIs, because both these APIs use some filters for searching results, so in this case, adding `django_filters` to the settings.py will resolve this issue. [Reference](https://github.com/carltongibson/django-filter/issues/562)
2. In the requirements.txt, replace the package `backports.zoneinfo==0.2.1` with `backports.zoneinfo;python_version<"3.9"`. Avoid installing this package for python versions > 3.9

Added:
1. new model (applicants) in `apps/jobs/models.py`
2. Create an action under the Job viewset to handle the apply functionality (initial)

### CODE REVIEW

File: apps/jobs/models.py [hunk 1]
```diff
@@ -4,6 +4,14 @@
 
 from apps.accounts.models import User as UserAuth
 
+STATUS_CHOICES = (
+    ("under-reviewed", "Under-Reviewed"),
```
Comment (HimanshuSharma-Angel) [inline apps/jobs/models.py:8]: status choice would be initiated

File: apps/jobs/models.py [hunk 2]
```diff
@@ -94,3 +103,22 @@ def save(self, *args, **kwargs):
...
+    """
+
+    class Meta:
+        db_table = "tbl_applicants"
+
+    job = models.ForeignKey(Job, on_delete=models.CASCADE, null=False)
+    user = models.ForeignKey(User, on_delete=models.CASCADE, null=False)
+    status = models.CharField(max_length=30, choices=STATUS_CHOICES, default="applied")
+    created_at = models.DateTimeField(auto_now_add=True)
+    resume = models.FileField(upload_to="resume/", null=True, blank=True)
```
Comment (HimanshuSharma-Angel) [inline apps/jobs/models.py:120]: we dont need it as we can fetch it from user table

**Review by HimanshuSharma-Angel — CHANGES_REQUESTED** [review]

File: apps/jobs/views.py [hunk 3]
```diff
@@ -141,6 +141,91 @@ def users(self, request, pk=None):
...
+    def apply(self, request, pk=None):
+        """Apply job functionality implementation"""
+
+        common_response_parameters = {
+            "status": status.HTTP_400_BAD_REQUEST,
+            "content_type": "application/json",
+        }
+
+        # Change this actions' method to GET later
+        if "user_id" not in request.data:
```
Comment (hims1911) [inline apps/jobs/views.py:154]: for time being its ok but in future will fetch from JWT

File: apps/jobs/views.py [hunk 4]
```diff
@@ -141,6 +141,91 @@ def users(self, request, pk=None):
...
+        # job-id
+        response = validate_id(job_id, "job-id", Job)
+        if response:
+            return response
+
+        # user-id
+        response = validate_id(user_id, "user-id", User)
+        if response:
+            return response
+
```
Comment (hims1911) [inline apps/jobs/views.py:189]: we can club both the statement.

File: apps/jobs/views.py [hunk 5]
```diff
@@ -141,6 +141,91 @@ def users(self, request, pk=None):
...
+                return Response(
+                    {"message": f"You don't have {key} updated"},
+                    **common_response_parameters,
+                )
+
+        # Prepare the overall dictionary to save into the database
+        # Add job-id, user-id to the applyjob_data
+        applyjob_data["job_id"] = job_id
+        applyjob_data["user_id"] = user_id
+
```
Comment (hims1911) [inline apps/jobs/views.py:219]: lets discuss this tomorrow.

**Review by hims1911 — CHANGES_REQUESTED** [review]

File: apps/jobs/views.py (hunk 4 above)
Comment (hims1911) [reply apps/jobs/views.py:189]: `if validate_id(job_id, "job-id", Job) and  response = validate_id(user_id, "user-id", User):
    return generic response() 
`

File: apps/jobs/views.py [hunk 6]
```diff
@@ -141,6 +141,91 @@ def users(self, request, pk=None):
...
+        # Get the data from the user's database (only resume N cover_letter)
+        applyjob_data = (
+            User.objects.filter(user_id=user_id)
+            .values("resume", "cover_letter")
+            .first()
+        )
+
+        for key, value in applyjob_data.items():
+            if not value:
```
Comment (hims1911) [inline apps/jobs/views.py:209]: optimise this

File: apps/jobs/views.py [hunk 7]
```diff
@@ -141,6 +141,91 @@ def users(self, request, pk=None):
...
+                    **common_response_parameters,
+                )
+
+        # Prepare the overall dictionary to save into the database
+        # Add job-id, user-id to the applyjob_data
+        applyjob_data["job_id"] = job_id
+        applyjob_data["user_id"] = user_id
+
+        # Add this application into the database
+        applyjob = Applicants(**applyjob_data)
```
Comment (hims1911) [inline apps/jobs/views.py:221]: default choice - initiated

hr update -> status 

applyjob_data["status"] = request.data["status"]

Comment (hims1911) [reply apps/jobs/views.py:221]: `if user_type is HR`

**Review by hims1911 — CHANGES_REQUESTED** [review]

**Pull request merged**